src/dispatch-payload.ts

```diff
@@ -142,3 +125,26 @@ function getFunctionCallbackID(
         ?.function_data?.function?.callback_id ?? "";
   }
 }
+
+export function extractBaseHandlerArgsFromPayload(
+  payload: InvocationPayload<ValidInvocationPayloadBody>,
+): BaseHandlerArgs {
+  const { body, context } = payload;
+  const env = context.variables || {};
+  const team_id = context.team_id || "";
+  const enterprise_id = body.enterprise_id ||
+    (body as BaseEventInvocationBody).enterprise?.id || "";
```

Comment:
I think the explicit casting here makes it a bit clearer why we look for the properties on the payloads in different areas: because the payloads differ based on the event.